src/components/PuppyDetail.jsx

```diff
@@ -0,0 +1,150 @@
+            {/* input holder */}
+            <div className="p-5 sm:w-96 w-full flex flex-col justify-start items-center">
+              <Input
+                placeholder="Amount (ETH)"
+                name="amount"
+                type="number"
+                handleChange={handleChange}
+              />
+              <Input
+                placeholder="Keyword (Gif)"
+                name="keyword"
+                type="text"
+                handleChange={handleChange}
+              />
+              <Input
+                placeholder="Enter Message"
+                name="message"
+                type="text"
+                handleChange={handleChange}
+              />
+
+              <div className="h-[1px] w-full bg-gray-400 my-2" />
+
+              {isLoading ? (
+                <Loader />
+              ) : (
+                <button
+                  type="button"
+                  onClick={handleSubmit}
+                  className="text-white w-full mt-2 border-[1px] p-2 border-[#3d4f7c] hover:bg-[#3d4f7c] rounded-full cursor-pointer"
+                >
+                  Send now
+                </button>
+              )}
+            </div>
+          </div>
+        </div>
+      </div>
+
+      {/* list of transaction */}
+      <table className="border-collapse border border-slate-400 text-white w-full my-5">
+        <thead>
+          <tr>
+            {['Donator', 'Receiver', 'Amount', 'Message', 'Transaction Time'].map((v, i) => (
+              <th key={i} className="border border-slate-300 py-2">
+                {v}
+              </th>
+            ))}
+          </tr>
+        </thead>
+        <tbody>
+          {transactions &&
+            transactions
+              .filter((x) => x.puppyId == params.id)
+              .reverse()
+              .map((transaction, i) => (
+                <tr key={i}>
+                  <td className="border border-slate-300 text-center py-1">
+                    <a
+                      href={`https://ropsten.etherscan.io/address/${transaction.addressFrom}`}
+                      target="_blank"
+                      rel="noreferrer"
+                    >
+                      <p className="text-white text-base hover:text-blue-400">
+                        {shortenAddress(transaction.addressFrom)}
+                      </p>
+                    </a>
+                  </td>
+                  <td className="border border-slate-300 text-center py-1">
+                    <a
+                      href={`https://ropsten.etherscan.io/address/${transaction.addressTo}`}
+                      target="_blank"
+                      rel="noreferrer"
+                    >
+                      <p className="text-white text-base hover:text-blue-400">
+                        {shortenAddress(transaction.addressTo)}
+                      </p>
+                    </a>
+                  </td>
```

Comment:
Remove "Receiver" column in puppy's donation table.

```suggestion
            {['Donator', 'Amount', 'Message', 'Transaction Time'].map((v, i) => (
              <th key={i} className="border border-slate-300 py-2">
                {v}
              </th>
            ))}
          </tr>
        </thead>
        <tbody>
          {transactions &&
            transactions
              .filter((x) => x.puppyId == params.id)
              .reverse()
              .map((transaction, i) => (
                <tr key={i}>
                  <td className="border border-slate-300 text-center py-1">
                    <a
                      href={`https://ropsten.etherscan.io/address/${transaction.addressFrom}`}
                      target="_blank"
                      rel="noreferrer"
                    >
                      <p className="text-white text-base hover:text-blue-400">
                        {shortenAddress(transaction.addressFrom)}
                      </p>
                    </a>
                  </td>
```

Reply:
Addressed in [b78f18f](https://github.com/lohsuan/puppy_sponsor/pull/7/commits/b78f18fb3daf5a701d0e36c4b31a77e4bde0d9cc).